src/visualizers/panels/ExecutionIndex/ExecutionIndexControl.js

```diff
@@ -38,45 +41,88 @@ define([
     };
 
     ExecutionIndexControl.prototype._initWidgetEventHandlers = function () {
-        this._widget.setExecutionDisplayed = this.setExecutionDisplayed.bind(this);
+        this._widget.setDisplayedExecutions = this.setDisplayedExecutions.bind(this);
     };
 
-    ExecutionIndexControl.prototype.setExecutionDisplayed = function (id, bool) {
-        var lines = this._linesForExecution[id] || [],
-            otherLines,
-            wasMultiLine = this.displayedExecCount() > 1,
-            isMultiLine;
+    ExecutionIndexControl.prototype.setDisplayedExecutions = function (beforeClickIds, afterClickIds) {
+        let addedExecutions = afterClickIds.filter(id => !beforeClickIds.includes(id));
+        let removedExecutions = beforeClickIds.filter(id => !afterClickIds.includes(id));
+        let added = addedExecutions.length > 0;
+        let updatedExecutions = added ? addedExecutions : removedExecutions;
+        this._updateGraphData(...updatedExecutions, added);
+    };
 
-        this._logger.info(`setting execution ${id} to ${bool ? 'displayed' : 'hidden'}`);
+    ExecutionIndexControl.prototype._updateGraphData = function (id, bool) {
         this.displayedExecutions[id] = bool;
-
-        // If we just crossed the multi line threshold, then update all the lines
-        isMultiLine = this.displayedExecCount() > 1;
-        if (isMultiLine !== wasMultiLine) {
-            // Refresh the other lines visible
-            otherLines = Object.keys(this.displayedExecutions)
-                .filter(eId => this.displayedExecutions[eId] && (eId !== id))
-                .map(id => this._linesForExecution[id] || [])
-                .reduce((l1, l2) => l1.concat(l2), []);
-
-            this._updateLines(otherLines, false);
-            this._updateLines(otherLines, true);
+        let displayedGraphExecIds = Object.keys(this.displayedExecutions)
+            .filter((id) => !!this.displayedExecutions[id]);
+        if (this._currentPlotsDataId) {
```

Comment:
Is this really needed? If PlotlyGraphWidget is simplified to simply pass the JSON state for the given node to the plotly library, then we may not need to call `removeNode` when we are simply going to be resetting it. This could simplify this method a bit (especially when combined with the comments about `_consolidateGraphData`.

Reply:
This method has been refactored to make `removeNode` when no execution is displayed.